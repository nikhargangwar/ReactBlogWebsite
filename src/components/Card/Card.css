
.cardMain {
    width: 245px;
    height: 350px;
    background-color: #fff;
    margin: 10px 0px;
    border: 0.5px solid rgb(156, 156, 156);
}

.cardImage {
    width: 100%;
    height: 60%;
    display: flex;
    justify-content: center;
    align-items: center;
    overflow: hidden
}

.cardImage img {
    width: 100%;
    height: auto;
    background-size: cover;
    background-repeat: no-repeat;
    background-position: center;
    flex-shrink: 0;
    min-width: 100%;
    min-height: 100%
}

.cardTime {
    height: 4%;
    width: 100%;
    display: flex;
    justify-content: space-between;
    padding: 0px5px;
    margin: 3px;
}

.cardTime h5 {
    margin: 0px 5px;
    padding: 0px;
    font-size: 10px;
    font-family: 'Roboto', sans-serif;
    color: rgb(137, 137, 137);

}

.cardTitle {
    height: 13%;
    width: 90%;
    display: flex;
    justify-content: center;
    align-items: center;

    padding: 0px 6px;

}

.cardTitle h3 {

    /* padding: 0px;
    font-size: 10px;
    font-weight: 100;
    font-family: 'Roboto', sans-serif;
    color: rgb(137, 137, 137); */
    overflow: scroll;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  /* truncate to 4 lines */
  -webkit-line-clamp: 2;

}

.cardDescription {
    height: 13%;
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;

}

.cardDescription h5 {
    margin: 0px 5px;
    padding: 0px;
    font-size: 10px;
    font-weight: 100;
    font-family: 'Roboto', sans-serif;
    color: rgb(104, 104, 104);
    overflow: scroll;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  /* truncate to 4 lines */
  -webkit-line-clamp: 3;

}

#cardLine {
    width: 90%;
    border: 0.1px solid rgb(0, 0, 0, 0.2);

    margin: 0px auto;
    margin-bottom: 2px;
}

.cardReaction {
    height: 6%;
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;

}

.cardReaction>img {
    height: 80%;
    margin: 0px 10px;
}

.clapCount {
    height: 20px;
    width: 50px;
    display: flex;
    align-items: center;
    color: rgb(137, 137, 137);
}

.clapCount>img {
    height: 100%;
    margin: 0px 10px;
}