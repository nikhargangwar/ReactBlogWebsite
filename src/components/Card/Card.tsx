import './Card.css';
// import cardImage from '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/images/';
import clapIcon from '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/Icons/clapping.svg';
import heartBankIcon from '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/Icons/heart-black.svg';
import heartRedIcon from '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/Icons/heart-red.svg';

import { useState } from 'react';

interface propTypes {
  date: string;
  readingTime: string;
  title: string;
  description: string;
  claps: number;
  liked: boolean;
  image: string;
}


function Card({date,readingTime,title,description,claps,liked,image}:propTypes){

const [countClap,setCountClap]= useState(claps);
const [isLiked,setIsLiked]=useState(liked)
const clickHandler =()=>{
  setCountClap(countClap+1);
}

// const imagePath =cardImage+image;

const heartClickHandler=()=>{
  setIsLiked(!isLiked);
}

console.log(image);

return(
    <div className="cardMain">
    <div className="cardImage">
      <img src={require(`../../assets/images/${image}`)} />
    </div>
    <div className="cardTime">
      <h5>{date}</h5>
      <h5>{readingTime}</h5>
    </div>
    <div className="cardTitle">
      <h3>{title}</h3>
    </div>
    <div className="cardDescription">
      <h5>
       {description}
      </h5>
    </div>
    <hr id="cardLine" />
    <div className="cardReaction">
      <div className="clapCount">
        <img onClick={clickHandler} src={clapIcon} />
        <h5>{countClap}</h5>
      </div>
      {isLiked ? <img onClick={heartClickHandler} src={heartRedIcon} alt="images" />:<img onClick={heartClickHandler} src={heartBankIcon} alt="images" />}
      
    </div>
  </div>
)
}

export default Card;