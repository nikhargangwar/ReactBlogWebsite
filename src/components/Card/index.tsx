import './Card.css';
// import cardImage from '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/images/';
import clapIcon from '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/Icons/clapping.svg';
import heartBankIcon from '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/Icons/heart-black.svg';
import heartRedIcon from '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/Icons/heart-red.svg';

import { useState } from 'react';

interface propTypes {
        date: string;
  readingTime: string;
  title: string;
  description: string;
  claps: number;
  liked: boolean;
  image: string;
}


function Card({date,readingTime,title,description,claps,liked,image}:propTypes){

const [countClap,setCountClap]= useState(claps);
const [isLiked,setIsLiked]=useState(liked);
const [flag,setFlag]= useState(false);

const clickHandler =()=>{
  setCountClap(countClap+1);
  // if(!flag)
  // {setCountClap(countClap+1);
  // setFlag(true);
  // }
  // else{
  //   setCountClap(countClap-1);
  //   setFlag(false);
  // }
   
}

// const imagePath =cardImage+image;

const heartClickHandler=()=>{
  setIsLiked(!isLiked);
}

console.log(image);

return(
    <div className="cardMain">
    <div className="cardImage">
      <img src={require(`../../assets/images/${image}`)} />
    </div>
    <div className="cardTime">
      <h5>{date}</h5>
      <h5>{readingTime}</h5>
    </div>
    <div className="cardTitle">
      <h3>{title}</h3>
    </div>
    <div className="cardDescription">
      <h5>
       {description}
      </h5>
    </div>
    <hr id="cardLine" />
    <div className="cardReaction">
      <div className="clapCount">
        <img className="clap-button"onClick={clickHandler} src={clapIcon} />
        <h5 className="clap">{countClap}</h5>
      </div>
      {isLiked ? <img className='like' onClick={heartClickHandler} src={heartRedIcon} alt="redHeart" />:<img className='like' onClick={heartClickHandler} src={heartBankIcon} alt="blackHeart" />}
      
    </div>
  </div>
)
}

export default Card;