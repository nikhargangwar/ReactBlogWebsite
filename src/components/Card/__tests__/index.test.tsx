import React from 'react';
import Card from  '..';
import { fireEvent, render } from '@testing-library/react';


describe('testing for card component',()=>{
    it('should toggle like on click', () => {
        const { container } = render(
          <Card
            image="abstract.png"
            date="2nd Januray, 2018"
            readingTime="2 mins"
            title="The future of abstract art and the culture ..."
            description="Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your..."
            claps={10}
            liked={false}
          />
        );
        expect(container).toMatchSnapshot();
       
      });
      it('should render correctly with liked', () => {
        const { container } = render(
          <Card
            image="abstract.png"
            date="2nd Januray, 2018"
            readingTime="2 mins"
            title="The future of abstract art and the culture ..."
            description="Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your..."
            claps={10}
            liked={true}
          />
        );
        expect(container).toMatchSnapshot();
      });
      it('should render correctly with 0 claps', () => {
        const { container } = render(
          <Card
            image="abstract.png"
            date="2nd Januray, 2018"
            readingTime="2 mins"
            title="The future of abstract art and the culture ..."
            description="Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your..."
            claps={0}
            liked={false}
          />
        );
        expect(container).toMatchSnapshot();
      });

      // One of the way to test
      it('should toggle like on click', () => {
        const { getByAltText } = render(
          <Card
            image="abstract.png"
            date="2nd Januray, 2018"
            readingTime="2 mins"
            title="The future of abstract art and the culture ..."
            description="Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your..."
            claps={10}
            liked={false}
          />
        );
        
      const getBlackHeart = () => getByAltText('blackHeart') as HTMLImageElement;
        const getRedHeart = () => getByAltText('redHeart')  as HTMLImageElement;
        expect(getBlackHeart().src).toContain('heart-black.svg');
        fireEvent.click(getBlackHeart());

        expect(getRedHeart().src).toContain('heart-red.svg');
        fireEvent.click(getRedHeart());
        expect(getBlackHeart().src).toContain('heart-black.svg');
      });

      it('should increment claps on click', () => {
        const { container } = render(
          <Card
            image="abstract.png"
            date="2nd Januray, 2018"
            readingTime="2 mins"
            title="The future of abstract art and the culture ..."
            description="Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your..."
            claps={10}
            liked={false}
          />
        );
        const clap = container.querySelector('.clap');
        const clapButton = container.querySelector('.clap-button');
        expect(clap.innerHTML).toContain('10');
        fireEvent.click(clapButton);
        expect(clap.innerHTML).toContain('11');
      });

      
      it('should decrement claps on second click', () => {
        const { container } = render(
          <Card
            image="abstract.png"
            date="2nd Januray, 2018"
            readingTime="2 mins"
            title="The future of abstract art and the culture ..."
            description="Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your..."
            claps={10}
            liked={false}
          />
        );
        const clap = container.querySelector('.clap');
        const clapButton = container.querySelector('.clap-button');
        expect(clap.innerHTML).toContain('10');
        fireEvent.click(clapButton);
        expect(clap.innerHTML).toContain('11');
        fireEvent.click(clapButton);
        expect(clap.innerHTML).toContain('12');
      });
    
    
    // it('should toggle like on click', () => {
    //     const { container } = render(
    //       <Card
    //         image="abstract.png"
    //         date="2nd Januray, 2018"
    //         readingTime="2 mins"
    //         title="The future of abstract art and the culture ..."
    //         description="Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your..."
    //         claps={10}
    //         liked={false}
    //       />
    //     );
       // expect(container).toMatchSnapshot();
        // const like = container.querySelector('.like');
        // expect(like).toHaveAttribute('src', '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/Icons/heart-black.svg');
        // fireEvent.click(like);
        // expect(like).toHaveAttribute('src', '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/Icons/heart-black.svg');
        // fireEvent.click(like);
        // expect(like).toHaveAttribute('src', '/Users/nikhar_gangwar/reactWebsitePractice/my-app/src/assets/Icons/heart-black.svg');



        // const like = container.queryBy('.like');


// console.log(like);
//        expect(like.src).toContain('heart-black.svg');
//         fireEvent.click(like[0]);
//         expect(like.src).toContain('heart-red.svg');
//         fireEvent.click(like);
//         expect(like.src).toContain('heart-black.svg');
//       });
      
})  