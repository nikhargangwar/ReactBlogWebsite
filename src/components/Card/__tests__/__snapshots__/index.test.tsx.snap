// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing for card component should render correctly with 0 claps 1`] = `
<div>
  <div
    class="cardMain"
  >
    <div
      class="cardImage"
    >
      <img
        src="abstract.png"
      />
    </div>
    <div
      class="cardTime"
    >
      <h5>
        2nd Januray, 2018
      </h5>
      <h5>
        2 mins
      </h5>
    </div>
    <div
      class="cardTitle"
    >
      <h3>
        The future of abstract art and the culture ...
      </h3>
    </div>
    <div
      class="cardDescription"
    >
      <h5>
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </h5>
    </div>
    <hr
      id="cardLine"
    />
    <div
      class="cardReaction"
    >
      <div
        class="clapCount"
      >
        <img
          class="clap-button"
          src="clapping.svg"
        />
        <h5
          class="clap"
        >
          0
        </h5>
      </div>
      <img
        alt="blackHeart"
        class="like"
        src="heart-black.svg"
      />
    </div>
  </div>
</div>
`;

exports[`testing for card component should render correctly with liked 1`] = `
<div>
  <div
    class="cardMain"
  >
    <div
      class="cardImage"
    >
      <img
        src="abstract.png"
      />
    </div>
    <div
      class="cardTime"
    >
      <h5>
        2nd Januray, 2018
      </h5>
      <h5>
        2 mins
      </h5>
    </div>
    <div
      class="cardTitle"
    >
      <h3>
        The future of abstract art and the culture ...
      </h3>
    </div>
    <div
      class="cardDescription"
    >
      <h5>
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </h5>
    </div>
    <hr
      id="cardLine"
    />
    <div
      class="cardReaction"
    >
      <div
        class="clapCount"
      >
        <img
          class="clap-button"
          src="clapping.svg"
        />
        <h5
          class="clap"
        >
          10
        </h5>
      </div>
      <img
        alt="redHeart"
        class="like"
        src="heart-red.svg"
      />
    </div>
  </div>
</div>
`;

exports[`testing for card component should toggle like on click 1`] = `
<div>
  <div
    class="cardMain"
  >
    <div
      class="cardImage"
    >
      <img
        src="abstract.png"
      />
    </div>
    <div
      class="cardTime"
    >
      <h5>
        2nd Januray, 2018
      </h5>
      <h5>
        2 mins
      </h5>
    </div>
    <div
      class="cardTitle"
    >
      <h3>
        The future of abstract art and the culture ...
      </h3>
    </div>
    <div
      class="cardDescription"
    >
      <h5>
        Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
      </h5>
    </div>
    <hr
      id="cardLine"
    />
    <div
      class="cardReaction"
    >
      <div
        class="clapCount"
      >
        <img
          class="clap-button"
          src="clapping.svg"
        />
        <h5
          class="clap"
        >
          10
        </h5>
      </div>
      <img
        alt="blackHeart"
        class="like"
        src="heart-black.svg"
      />
    </div>
  </div>
</div>
`;
